### ISSUE TITLE
wrapping HTML output (blaze-html dependency)

### ISSUE THREAD

**Issue by reagle** [jgm/pandoc#7764]
The [documentation](https://pandoc.org/MANUAL.html#general-writer-options) presently says "Automatic wrapping does not currently work in HTML output." @jgm responds on this list "It's because we use an external library (blaze-html) to produce the HTML, and it doesn't do wrapping."

I [don't see this issue documented on the blaze repo](https://github.com/jaspervdj/blaze-html/issues), should it?

**Comment by jgm** [comment]
I don't think many people who generate HTML care about this feature.

We could, of course, stop using blaze and use our own XML generation tools (as we do for JATS and Docbook), which would give us wrapping.

But:

- that would require rewriting the HTML writer, quite a bit of work
- it would require a breaking API change, since we'd no longer be able to export HTML writer functinos that produce blaze Html structures instead of text
- people who rely on these functions might well complain
- the writer would also become slower

So, I'd say it's not worth doing that.

A better option would be to write our own blaze renderer (converting a blaze Html struture to Text); we could then make it do wrapping.  That doesn't look terribly hard; here's an example of a renderer:
https://hackage.haskell.org/package/blaze-markup-0.8.2.8/docs/src/Text.Blaze.Renderer.Pretty.html#renderMarkup

**Comment by jgm** [comment]
I guess the easiest approach would be to create a function
```hs
renderMarkup :: Markup -> Doc Text
```
and then use doclayout's renderer to handle the wrapping.
We could base it on
https://hackage.haskell.org/package/blaze-markup-0.8.2.8/docs/src/Text.Blaze.Renderer.Text.html#renderMarkup

**Comment by reagle** [comment]
I agree that this isn't a big use case and there are utils out there such as HTML Tidy that can wrap. However, they have their own issues and when rereading the docs,  "does not currently work " made me think something is temporarily broken. I then couldn't find an issue. Perhaps the doc could be changed to say: "Automatic wrapping is not supported in HTML output." If there is an easy way to support wrapping, all the better. :)

**Comment by jgm** [comment]
I have a PR now that addresses this issue.

Potential downsides:

- changes default behavior (since `--wrap=auto` is the default for all formats); HTML that wasn't wrapped before will now wrap
- performance is slightly worse with wrapping

I think it's probably worth merging this, but wondering if anyone has comments.
